src/main/java/se/sundsvall/casedata/integration/messageexchange/configuration/MessageExchangeConfiguration.java

```diff
@@ -24,9 +28,17 @@ JsonFormWriter jsonFormWriter() {
 	FeignBuilderCustomizer feignBuilderCustomizer(final ClientRegistrationRepository clientRepository, final MessageExchangeProperties messageExchangeProperties) {
 		return FeignMultiCustomizer.create()
 			.withErrorDecoder(new ProblemErrorDecoder(CLIENT_ID))
+			.withRequestInterceptor(builder -> builder.header(Identifier.HEADER_NAME, createSentByHeaderValue(Identifier.get())))
 			.withRequestTimeoutsInSeconds(messageExchangeProperties.connectTimeout(), messageExchangeProperties.readTimeout())
 			.withRetryableOAuth2InterceptorForClientRegistration(clientRepository.findByRegistrationId(CLIENT_ID))
 			.composeCustomizersToOne();
 	}
 
+	String createSentByHeaderValue(final Identifier identifier) {
+		return ofNullable(identifier)
+			.filter(i -> allNotNull(i.getType(), i.getValue()))
```

Comment:
You don't need to check this. If type and/or value is null, then toHeaderValue() will also be null.